### ISSUE TITLE
private.php and $lang->comma

### ISSUE THREAD

**Issue by frostschutz** [mybb/mybb#3038]
If $lang->comma is set to anything other than the ASCII comma ',' - which may be the case in asian language which has a different comma character '、', the private message system, specifically replying to multiple users, stops working.

![comma](https://user-images.githubusercontent.com/194658/36432442-82e54a02-165a-11e8-967e-b6639c50a930.png)

The recipients strings are imploded/joined with $lang->comma, but exploded/split with regular , so a list of multiple usernames is mis-interpreted as a single user name instead.

Perhaps this userlist parsing should use a regular expression which uses both regular comma , hardcoded, plus whatever $lang->comma is set to.

To reproduce, just edit global.lang.php and replace $l['comma']=","; with any other character instead of the actual comma , .

**Comment by effone** [comment]
@frostschutz why a regular comma is required to be hardcoded? Is it because of being comma as vastly used for separators? Also it is a big problem if a username contains some character or symbol same as set to $lang->comma.

**Comment by frostschutz** [comment]
Language isn't fixed, so a forum may have multiple languages installed, and every user can set their preferred language in User CP options.

So $lang->comma for User A might be using , user B 、 user C ・ user D ／ - I don't know what languages are out there and what comma characters they may be using.

To make it work for everyone either you have to save as an array in the first place and display in native $lang->comma for everyone and/or simply declare that , is not language but syntax here. Then , is hard coded for separating user names. No language involved.

It works like that for bbcode too, [quote] is syntax regardless of other languages using a different character set such as 「ｑｕｏｔｅ」.

**Comment by effone** [comment]
So, if the scenario is dependent on variable only dependent on $lang->comma, lets drop the intention to hardcode regular comma and drive the scenario with $lang->comma only.
Talking about coding: for `preg_split( )` is it okay to use variable in delimiter? We have to repass the list with several explodes.
Alternate thought is we can also go for identifying non-allowed username characters and split as per.
I have a modification ready to PR with applying separator as $lang->comma only. Waiting for the decision as I am not sure about the best scenario. As I have mentioned earlier also, it will be a good choice to restrict $lang->comma value to be a list of all possible language separators.

**Comment by Sama34** [comment]
I agree this use of the comma shouldn't be. And we should scrap that line altogether. You can't even put a comma in there without it being considered part of the username anyways.
